js/app/src/components/StatusTracker/ToastContent.svelte

```diff
@@ -0,0 +1,173 @@
+<script lang="ts">
+	import { Alert, Information, Warning } from "@gradio/icons";
+	import { createEventDispatcher, onMount, afterUpdate } from "svelte";
+	import { fade } from "svelte/transition";
+	import type { ThemeMode } from "../types";
+
+	export let message: string = "";
+	export let description: string = "";
+	export let type: string = "";
+	export let theme_color: string = "";
+	export let theme_mode: ThemeMode;
+	export let id: number;
+
+	const dispatch = createEventDispatcher();
+
+	function close_message() {
+		dispatch("close", id);
+	}
+
+	let lightStyles = {
+		"toast-bg-color": `var(--color-${theme_color}-50)`,
+		"toast-text-color": `var(--color-${theme_color}-700)`,
+		"toast-border-color": `var(--color-${theme_color}-700)`,
+		"toast-icon-color": `var(--color-${theme_color}-700)`
+	};
+
+	let darkStyles = {
+		"toast-bg-color": `var(--color-grey-950)`,
+		"toast-text-color": `var(--color-${theme_color}-50)`,
+		"toast-border-color": `var(--color-${theme_color}-500)`,
+		"toast-icon-color": `var(--color-${theme_color}-500)`
+	};
+
+	let cssVarStyles = "";
+
+	function updateCssVarStyles() {
+		cssVarStyles = Object.entries(
+			theme_mode === "light" ? lightStyles : darkStyles
+		)
+			.map(([key, value]) => `--${key}:${value}`)
+			.join(";");
+	}
+
+	afterUpdate(updateCssVarStyles);
+
+	onMount(() => {
+		setTimeout(() => {
+			close_message();
+		}, 10000);
+	});
+
+	$: updateCssVarStyles();
+</script>
+
+<div
+	style={cssVarStyles}
```

Comment:
I think it might be more readable if, instead of merging everything into one style string, we keep each property separate, e.g.
```svelte
<div
    style:background-color={theme["toast-bg-color"]}
    style:color={theme["toast-text-color"]}
    ...
>
```
And the theme variable will just be either lightStyles or darkStyles depending on theme_mode